types/es-abstract/2015/DefinePropertyOrThrow.d.ts

```diff
@@ -0,0 +1,4 @@
+import type { PropertyDescriptor, PropertyKey } from '../index';
+
+declare function DefinePropertyOrThrow(O: object, P: PropertyKey, desc: PropertyDescriptor & ThisType<any>): boolean;
```

Comment:
can you explain the `ThisType<any>`?

Reply:
That’s a **TypeScript** marker interface that tells the **TypeScript** compiler that the inferred `this` type of function object properties shouldn’t be the enclosing object (which in this case is `PropertyDescriptor`), but is instead the type parameter of `ThisType`.